microsoft/fluentui-apple
# [vNext Tokens] `state` -> `configuration`

### Platforms Impacted

### Description of changes

As part of our effort to consolidate and update our approach to our new Fluent controls, we've decided to formally swap the name for our SwiftUI controls' internal data structure from `state` to `configuration`.

The reasons for this change:
* `state` is a highly overloaded term in Apple-land, from `@State` in SwiftUI to `UIControl.State` for the interaction state of the control (e.g. highlighted, disabled), there is potential for confusion and conflict.
* On the other hand, `configuration` is only lightly used in either SwiftUI or UIKit, and only for buttons so far.
* `configuration` also more precisely describes the goal of the class/property: to describe how the control is "configured". This can include titles or images in buttons, or the sub-items in an aggregate control like `AvatarGroup`, along with custom token overrides for every other control. 

As for why now: this branch will result in a fair bit of churn to our clients when it merges, so might as well pull the bandage off now!

### Verification

Since this is a fairly wide-ranging change, the testing approach was more "bug bash" style than particularly focused on one area. My verification flow:
* Ensure every existing vNext control still displays in its demo controller.
* Verify the SwiftUI versions of vNext controls with support for that hosting environment.
* Verify custom tokens for controls with this view hooked up.

### Pull request checklist

This PR has considered:
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/939)
